src/selector.rs

```diff
@@ -58,13 +68,29 @@ impl parser::SelectorImpl for Simple {
 
     type NonTSPseudoClass = NonTSPseudoClass;
     type PseudoElement = PseudoElement;
+
+    // see: https://github.com/servo/servo/pull/19747#issuecomment-357106065
+    type ExtraMatchingData = String;
+
+    fn is_active_or_hover(pc: &NonTSPseudoClass) -> bool {
+        matches!(*pc, NonTSPseudoClass::Active | NonTSPseudoClass::Hover)
+    }
 }
 
 /// Non Tree-Structural Pseudo-Class.
 #[derive(Debug, Clone, Copy, PartialEq, Eq)]
-pub enum NonTSPseudoClass {}
+pub enum NonTSPseudoClass {
+    /// Unvisited links
+    Link,
+    /// Visited links
+    Visited,
+    /// user hovers
+    Hover,
+    /// active links
+    Active,
+}
```

Comment:
Is there a point in adding these? They're irrelevant in the context of HTML scraping.

Reply:
Considering that nothing ever creates `Active` or `Hover`, `is_active_or_hover` could just always return false?

Reply:
ok, I think I see your point. We have a "Simple" implementation of the parser, it's not really needed at this time.